homeassistant/components/voice_assistant/pipeline.py

```diff
@@ -0,0 +1,126 @@
+"""Classes for voice assistant pipelines."""
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Callable
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Any
+
+from homeassistant.components import conversation
+from homeassistant.core import Context, HomeAssistant
+from homeassistant.util.dt import utcnow
+
+DEFAULT_TIMEOUT = 30  # seconds
```

Comment:
Should we reduce this to 15? 30 is long.

Reply:
If you have local STT + TTS on a Pi 4, 15 seconds may not be enough time. We might want to may this a config option.